testworkload/testworkload.go

```diff
@@ -0,0 +1,274 @@
+package testworkload
+
+import (
+	"fmt"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	utilrand "k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+	batchv1 "k8s.io/client-go/pkg/apis/batch/v1"
+	extensionsv1beta1 "k8s.io/client-go/pkg/apis/extensions/v1beta1"
+)
+
+const (
+	pollTimeout  = 2 * time.Minute
+	pollInterval = 5 * time.Second
+)
+
+// TestWorkload creates a temp nginx deployment/service pair
+// that can be used as a test workload
+type TestWorkload struct {
```

Comment:
this should be still named as nginx.

so it will look like testworkload.Ngnix // NewNginx creates a temp nginx ....